store/iavl/store.go

```diff
@@ -31,25 +30,24 @@ var (
 // Store Implements types.KVStore and CommitKVStore.
 type Store struct {
 	tree Tree
-
-	// How many old versions we hold onto.
-	// A value of 0 means keep no recent states.
-	numRecent int64
-
-	// This is the distance between state-sync waypoint states to be stored.
-	// See https://github.com/tendermint/tendermint/issues/828
-	// A value of 1 means store every state.
-	// A value of 0 means store no waypoints. (node cannot assist in state-sync)
-	// By default this value should be set the same across all nodes,
-	// so that nodes can know the waypoints their peers store.
-	storeEvery int64
 }
 
 // LoadStore returns an IAVL Store as a CommitKVStore. Internally it will load the
 // store's version (id) from the provided DB. An error is returned if the version
 // fails to load.
 func LoadStore(db dbm.DB, id types.CommitID, pruning types.PruningOptions, lazyLoading bool) (types.CommitKVStore, error) {
-	tree, err := iavl.NewMutableTree(db, defaultIAVLCacheSize)
+	var iavlOpts *iavl.Options
+	if pruning.KeepEvery() == 0 && pruning.KeepRecent() == 0 {
```

Comment:
@AdityaSripal, I assume `iavl.DefaultOptions()` is semantically equivalent to the SDK's `PruneEverything` strategy? We should instead change `PruneEverything` to be the same then instead of doing this hacky conditional.

Reply:
Actually, reading the [doc](https://github.com/tendermint/iavl/blob/master/docs/tree/PRUNING.md) here seems to tell that there is no "prune everything" option. Am I reading that correctly?